common/metrics/opentelemetry_scope.go

```diff
@@ -93,20 +95,25 @@ func (m *opentelemetryScope) UpdateGauge(id int, value float64) {
 
 func (m *opentelemetryScope) StartTimer(id int) Stopwatch {
 	def := m.defs[id]
+	unit := def.unit
+	if len(unit) == 0 {
+		unit = Milliseconds
+	}
```

Comment:
is this only possible if we forget to set the unit for timer metrics in the defs.go? If we make sure all timer is created with the proposed newTimer() method, would we be able to remove this check?